components/memory/component.go

```diff
@@ -171,3 +150,150 @@ func (c *component) RegisterCollectors(reg *prometheus.Registry, dbRW *sql.DB, d
 	c.gatherer = reg
 	return metrics.Register(reg, dbRW, dbRO, tableName)
 }
+
+// CheckOnce checks the current pods
+// run this periodically
+func (c *component) CheckOnce() {
+	log.Logger.Infow("checking memory")
+	d := Data{
+		ts: time.Now().UTC(),
+	}
+	metrics.SetLastUpdateUnixSeconds(float64(d.ts.Unix()))
+	defer func() {
+		c.lastMu.Lock()
+		c.lastData = &d
+		c.lastMu.Unlock()
+	}()
+
+	cctx, ccancel := context.WithTimeout(c.ctx, 5*time.Second)
+	vm, err := mem.VirtualMemoryWithContext(cctx)
+	ccancel()
+	if err != nil {
+		d.err = err
+		return
+	}
+
+	d.TotalBytes = vm.Total
+	d.TotalHumanized = humanize.Bytes(vm.Total)
```

Comment:
Why generate human-friendly information here? Who uses the data and who is responsible for formatting it